pkg/fileutil/files.go

```diff
@@ -244,58 +257,77 @@ func CreateFile(p string) (*os.File, error) {
 	return os.Create(p)
 }
 
-func UploadFile(filePath, targetURL, formFileFieldName string, headers map[string]string) error {
-	file, err := os.Open(filePath)
+func UploadFile(args *UploadFileArguments) error {
+	file, err := os.Open(args.FilePath)
 	if err != nil {
 		return fmt.Errorf("error opening file: %w", err)
 	}
 	defer file.Close()
 
-	// Create a new buffer to store the multipart/form-data request
-	body := &bytes.Buffer{}
-	writer := multipart.NewWriter(body)
+	retryDelayInMS := args.InitialDelayInMS
+	currentUploadError := err
 
-	// Create a form field for the file
-	fileWriter, err := writer.CreateFormFile(formFileFieldName, filePath)
-	if err != nil {
-		return fmt.Errorf("error creating form file: %w", err)
-	}
-
-	// Copy the file content into the form field
-	_, err = ioCopy(fileWriter, file)
-	if err != nil {
-		return fmt.Errorf("error copying file content: %w", err)
-	}
+	for i := 1; i <= args.MaxTries; i++ {
+		// exponential backoff
+		time.Sleep(time.Duration(retryDelayInMS) * time.Millisecond)
```

Comment:
why do we need to sleep on the first try itself 😕?
Also, might be good to log that we would be sleeping for x seconds before retrying

Reply:
1st point I'll address.
2nd point: we discussed it but decided that we don't want to expose any extra logs to the user.

Reply:
IMHO showing a `Retrying...` type of message to the user would be useful.

Reply:
@danielhoherd We are already showing "please wait, attempting to upload the dags"

Reply:
Just curious...why 7 times?

Also wondering if for whatever reason the api returns 500, does it makes sense to try 7 times per deploy, per deployment? Would this be generating a huge volume of alerts for us?

Reply:
@kushalmalani It's 7 times because as per @pgvishnuram , dag-server should come online in about 2 minutes. Also, we don't want to keep the user waiting for longer.
We don't have any alerting on the dag-server component.